src/VirtoCommerce.Platform.Web/wwwroot/js/app/modularity/modularity.js

```diff
@@ -129,10 +130,99 @@ angular.module('platformWebApp').config(['$stateProvider', function ($stateProvi
             });
         }
 
+        function performAction(action, selection, blade, useParentBlade) {
+            if (_.any(selection)) {
+                blade.isLoading = true;
+
+                // eliminate duplicating nodes, if any
+                var grouped = _.groupBy(selection, 'id');
+                selection = [];
+                _.each(grouped, function (vals) {
+                    selection.push(_.last(vals));
+                });
+
+                // find not installed versions
+                if (action === 'update') {
+                    var installed = angular.copy(selection);
+                    selection = [];
+                    var modules = this.modules;
+                    _.each(installed, function (x) {
+                        var notInstalled = _.last(_.where(modules, { id: x.id, isInstalled: false }));
+                        selection.push(notInstalled);
+                    });
+                }
+
+                var modulesApiMethod = action === 'uninstall' ? modulesApi.getDependents : modulesApi.getDependencies;
+
+                modulesApiMethod(selection, function (data) {
+                    blade.isLoading = false;
+
+                    var dialog = {
+                        id: "confirm",
+                        action: action,
+                        selection: selection,
+                        dependencies: data,
+                        callback: function (resume) {
+                            if (resume) {
+                                blade.isLoading = true;
+
+                                // combine selected modules with dependencies/dependents
+                                _.each(selection, function (x) {
+                                    if (!_.findWhere(data, { id: x.id })) {
+                                        data.push(x);
+                                    }
+                                });
+
+                                // choose the right API method
```

Comment:
extract to function for better readability
function can return an object like { action, title }